src/main/java/org/datadog/jmxfetch/RemoteConnection.java

```diff
@@ -27,6 +29,8 @@
     private static final String KEY_STORE_PASSWORD_KEY = "key_store_password";
     private static final String DEFAULT_RMI_RESPONSE_TIMEOUT =
             "15000"; // Match the collection period default
+    // Match the default RMI connection timeout value
+    private static final Integer DEFAULT_RMI_CONNECTION_TIMEOUT = Integer.valueOf(0);
```

Comment:
Hi @prognant, I'm not sure if we should put 0 here as the default since 0 means infinite and the purpose of this PR is to put a non-zero default value for the socket timeout..

Reply:
@HantingZhang2  I agree, I'm always a little bit concerned about changing a default value, in this cas we may use something like 15sec (as initially suggested) or 30sec it should not impact existing deployment.